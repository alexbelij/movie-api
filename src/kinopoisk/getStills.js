/* @flow */

import R from 'ramda';

import { normalizeCdnImageUrl } from './utils';
import connector from './connector';

type KinopoiskApi$GalleryItem = {
  image: string,
  preview: string,
};
type KinopoiskApi$GetGalleryResponse = {
  gallery?: {
    kadr?: ?Array<KinopoiskApi$GalleryItem>,
    kadr_sp?: ?Array<KinopoiskApi$GalleryItem>,
    poster?: ?Array<KinopoiskApi$GalleryItem>,
  },
};

const getStills = async (id: number): Promise<?Array<string>> => {
  const res: ?KinopoiskApi$GetGalleryResponse =
    await connector.get('getGallery', { filmID: id });

  if (!res) return null;

  return R.pipe(
    R.propOr([], 'kadr'),
    R.map(
      ({ image }: KinopoiskApi$GalleryItem) => normalizeCdnImageUrl(image),
    ),
  )(res.gallery);
};

export default getStills;
