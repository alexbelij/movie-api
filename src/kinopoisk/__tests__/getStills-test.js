/* @flow */

import getStills from '../getStills';

const kpId = 714888; // Star Wars: The Force Awakens

describe('kinopoisk/getStills', () => {
  it('fetches movie stills from kinopoisk for a given id', async () => {
    expect(await getStills(kpId)).toMatchSnapshot();
  });
});
