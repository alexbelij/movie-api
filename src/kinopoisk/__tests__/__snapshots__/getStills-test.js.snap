exports[`getStills fetches movie stills from kinopoisk for a given id 1`] = `
Array [
  "http://st.kp.yandex.net/images/kadr/2751407.jpg",
  "http://st.kp.yandex.net/images/kadr/2751406.jpg",
  "http://st.kp.yandex.net/images/kadr/2751405.jpg",
  "http://st.kp.yandex.net/images/kadr/2751404.jpg",
  "http://st.kp.yandex.net/images/kadr/2751403.jpg",
  "http://st.kp.yandex.net/images/kadr/2751402.jpg",
  "http://st.kp.yandex.net/images/kadr/2751401.jpg",
  "http://st.kp.yandex.net/images/kadr/2751400.jpg",
  "http://st.kp.yandex.net/images/kadr/2751399.jpg",
  "http://st.kp.yandex.net/images/kadr/2751390.jpg",
  "http://st.kp.yandex.net/images/kadr/2751389.jpg",
  "http://st.kp.yandex.net/images/kadr/2751388.jpg",
  "http://st.kp.yandex.net/images/kadr/2735779.jpg",
  "http://st.kp.yandex.net/images/kadr/2735778.jpg",
  "http://st.kp.yandex.net/images/kadr/2735777.jpg",
  "http://st.kp.yandex.net/images/kadr/2735776.jpg",
  "http://st.kp.yandex.net/images/kadr/2735775.jpg",
  "http://st.kp.yandex.net/images/kadr/2735774.jpg",
  "http://st.kp.yandex.net/images/kadr/2735773.jpg",
  "http://st.kp.yandex.net/images/kadr/2735772.jpg",
  "http://st.kp.yandex.net/images/kadr/2735771.jpg",
  "http://st.kp.yandex.net/images/kadr/2735770.jpg",
  "http://st.kp.yandex.net/images/kadr/2735769.jpg",
  "http://st.kp.yandex.net/images/kadr/2735768.jpg",
  "http://st.kp.yandex.net/images/kadr/2735767.jpg",
  "http://st.kp.yandex.net/images/kadr/2735766.jpg",
  "http://st.kp.yandex.net/images/kadr/2735765.jpg",
  "http://st.kp.yandex.net/images/kadr/2735764.jpg",
  "http://st.kp.yandex.net/images/kadr/2735763.jpg",
  "http://st.kp.yandex.net/images/kadr/2735762.jpg",
  "http://st.kp.yandex.net/images/kadr/2704437.jpg",
  "http://st.kp.yandex.net/images/kadr/2704436.jpg",
  "http://st.kp.yandex.net/images/kadr/2704435.jpg",
  "http://st.kp.yandex.net/images/kadr/2704434.jpg",
  "http://st.kp.yandex.net/images/kadr/2704433.jpg",
  "http://st.kp.yandex.net/images/kadr/2704432.jpg",
  "http://st.kp.yandex.net/images/kadr/2704431.jpg",
  "http://st.kp.yandex.net/images/kadr/2704430.jpg",
  "http://st.kp.yandex.net/images/kadr/2696992.jpg",
  "http://st.kp.yandex.net/images/kadr/2696371.jpg",
  "http://st.kp.yandex.net/images/kadr/2696370.jpg",
  "http://st.kp.yandex.net/images/kadr/2696369.jpg",
  "http://st.kp.yandex.net/images/kadr/2696368.jpg",
  "http://st.kp.yandex.net/images/kadr/2696367.jpg",
  "http://st.kp.yandex.net/images/kadr/2696366.jpg",
  "http://st.kp.yandex.net/images/kadr/2696365.jpg",
  "http://st.kp.yandex.net/images/kadr/2696364.jpg",
  "http://st.kp.yandex.net/images/kadr/2694487.jpg",
  "http://st.kp.yandex.net/images/kadr/2693086.jpg",
  "http://st.kp.yandex.net/images/kadr/2693085.jpg",
  "http://st.kp.yandex.net/images/kadr/2693084.jpg",
  "http://st.kp.yandex.net/images/kadr/2693083.jpg",
  "http://st.kp.yandex.net/images/kadr/2693082.jpg",
  "http://st.kp.yandex.net/images/kadr/2693080.jpg",
  "http://st.kp.yandex.net/images/kadr/2689711.jpg",
  "http://st.kp.yandex.net/images/kadr/2689496.jpg",
  "http://st.kp.yandex.net/images/kadr/2689492.jpg",
  "http://st.kp.yandex.net/images/kadr/2687963.jpg",
  "http://st.kp.yandex.net/images/kadr/2687962.jpg",
  "http://st.kp.yandex.net/images/kadr/2687961.jpg",
  "http://st.kp.yandex.net/images/kadr/2687960.jpg",
  "http://st.kp.yandex.net/images/kadr/2687959.jpg",
  "http://st.kp.yandex.net/images/kadr/2687958.jpg",
  "http://st.kp.yandex.net/images/kadr/2687957.jpg",
  "http://st.kp.yandex.net/images/kadr/2687956.jpg",
  "http://st.kp.yandex.net/images/kadr/2687955.jpg",
  "http://st.kp.yandex.net/images/kadr/2687954.jpg",
  "http://st.kp.yandex.net/images/kadr/2687953.jpg",
  "http://st.kp.yandex.net/images/kadr/2687952.jpg",
  "http://st.kp.yandex.net/images/kadr/2687951.jpg",
  "http://st.kp.yandex.net/images/kadr/2687950.jpg",
  "http://st.kp.yandex.net/images/kadr/2687949.jpg",
  "http://st.kp.yandex.net/images/kadr/2687948.jpg",
  "http://st.kp.yandex.net/images/kadr/2687947.jpg",
  "http://st.kp.yandex.net/images/kadr/2687946.jpg",
  "http://st.kp.yandex.net/images/kadr/2687945.jpg",
  "http://st.kp.yandex.net/images/kadr/2687944.jpg",
  "http://st.kp.yandex.net/images/kadr/2687943.jpg",
  "http://st.kp.yandex.net/images/kadr/2687942.jpg",
  "http://st.kp.yandex.net/images/kadr/2687941.jpg",
  "http://st.kp.yandex.net/images/kadr/2687940.jpg",
  "http://st.kp.yandex.net/images/kadr/2687939.jpg",
  "http://st.kp.yandex.net/images/kadr/2687938.jpg",
  "http://st.kp.yandex.net/images/kadr/2687937.jpg",
  "http://st.kp.yandex.net/images/kadr/2687936.jpg",
  "http://st.kp.yandex.net/images/kadr/2683178.jpg",
  "http://st.kp.yandex.net/images/kadr/2681409.jpg",
  "http://st.kp.yandex.net/images/kadr/2665335.jpg",
  "http://st.kp.yandex.net/images/kadr/2665334.jpg",
  "http://st.kp.yandex.net/images/kadr/2665333.jpg",
  "http://st.kp.yandex.net/images/kadr/2665332.jpg",
  "http://st.kp.yandex.net/images/kadr/2665331.jpg",
  "http://st.kp.yandex.net/images/kadr/2665330.jpg",
  "http://st.kp.yandex.net/images/kadr/2665329.jpg",
  "http://st.kp.yandex.net/images/kadr/2665328.jpg",
  "http://st.kp.yandex.net/images/kadr/2665327.jpg",
  "http://st.kp.yandex.net/images/kadr/2665326.jpg",
  "http://st.kp.yandex.net/images/kadr/2665325.jpg",
  "http://st.kp.yandex.net/images/kadr/2665324.jpg",
  "http://st.kp.yandex.net/images/kadr/2665323.jpg",
  "http://st.kp.yandex.net/images/kadr/2665322.jpg",
  "http://st.kp.yandex.net/images/kadr/2665321.jpg",
  "http://st.kp.yandex.net/images/kadr/2664388.jpg",
  "http://st.kp.yandex.net/images/kadr/2646409.jpg",
  "http://st.kp.yandex.net/images/kadr/2646408.jpg",
  "http://st.kp.yandex.net/images/kadr/2631723.jpg",
  "http://st.kp.yandex.net/images/kadr/2631722.jpg",
  "http://st.kp.yandex.net/images/kadr/2631721.jpg",
  "http://st.kp.yandex.net/images/kadr/2631720.jpg",
  "http://st.kp.yandex.net/images/kadr/2631719.jpg",
  "http://st.kp.yandex.net/images/kadr/2631718.jpg",
  "http://st.kp.yandex.net/images/kadr/2631717.jpg",
  "http://st.kp.yandex.net/images/kadr/2631716.jpg",
  "http://st.kp.yandex.net/images/kadr/2631715.jpg",
  "http://st.kp.yandex.net/images/kadr/2601907.jpg",
  "http://st.kp.yandex.net/images/kadr/2583585.jpg",
  "http://st.kp.yandex.net/images/kadr/2583584.jpg",
  "http://st.kp.yandex.net/images/kadr/2578266.jpg",
  "http://st.kp.yandex.net/images/kadr/2578265.jpg",
  "http://st.kp.yandex.net/images/kadr/2578264.jpg",
  "http://st.kp.yandex.net/images/kadr/2578263.jpg",
  "http://st.kp.yandex.net/images/kadr/2578262.jpg",
  "http://st.kp.yandex.net/images/kadr/2578261.jpg",
  "http://st.kp.yandex.net/images/kadr/2578260.jpg",
  "http://st.kp.yandex.net/images/kadr/2578259.jpg",
  "http://st.kp.yandex.net/images/kadr/2578258.jpg",
  "http://st.kp.yandex.net/images/kadr/2578257.jpg",
  "http://st.kp.yandex.net/images/kadr/2578256.jpg",
  "http://st.kp.yandex.net/images/kadr/2578255.jpg",
  "http://st.kp.yandex.net/images/kadr/2578254.jpg",
  "http://st.kp.yandex.net/images/kadr/2578253.jpg",
  "http://st.kp.yandex.net/images/kadr/2578252.jpg",
  "http://st.kp.yandex.net/images/kadr/2530193.jpg",
  "http://st.kp.yandex.net/images/kadr/2517360.jpg",
  "http://st.kp.yandex.net/images/kadr/2517359.jpg",
  "http://st.kp.yandex.net/images/kadr/2517358.jpg",
  "http://st.kp.yandex.net/images/kadr/2517357.jpg",
]
`;
